python/reduction/definitions.py

```diff
@@ -15,6 +15,126 @@
 # TODO: Orthogonal configuration object.
 avx512 = True
 
+################################################################################
+### Reduction 1D
+################################################################################
+#   Op def: ( m )
+#    Iters: ({Red()})
+#             I   O
+#   Layout: {{m}, {}}
+
+
+# TODO: fold OpDSL definition and inferences into ProblemDefinition.
+@linalg_structured_op
+def reduction_1d(A=TensorDef(T, S.M), B=TensorDef(T, output=True)):
```

Comment:
If you consider adding another one, it would prob time to generalize this on the model of what @ftynse did on other things.